tests/hazmat/primitives/test_dh.py

```diff
@@ -74,31 +84,268 @@ def test_dh_numbers():
+
+        assert isinstance(parameters, dh.DHParametersWithSerialization)
+        parameter_numbers = parameters.parameter_numbers()
+        assert isinstance(parameter_numbers, dh.DHParameterNumbers)
+        assert bit_length(parameter_numbers.p) == key_size
+
+        assert isinstance(public, dh.DHPublicKeyWithSerialization)
+        assert isinstance(public.public_numbers(), dh.DHPublicNumbers)
+        assert isinstance(public.parameters(), dh.DHParameters)
+
+        assert isinstance(key, dh.DHPrivateKeyWithSerialization)
+        assert isinstance(key.private_numbers(), dh.DHPrivateNumbers)
+        assert isinstance(key.parameters(), dh.DHParameters)
+
+    def test_exchange(self, backend):
+        parameters = dh.generate_parameters(2, 512, backend)
+        assert isinstance(parameters, dh.DHParameters)
+
+        key1 = parameters.generate_private_key()
+        key2 = parameters.generate_private_key()
+
+        symkey1 = key1.exchange(key2.public_key())
+        assert symkey1
+        assert len(symkey1) == 512 // 8
+
+        symkey2 = key2.exchange(key1.public_key())
+        assert symkey1 == symkey2
+
+    def test_exchange_algorithm(self, backend):
+        parameters = dh.generate_parameters(2, 512, backend)
+
+        key1 = parameters.generate_private_key()
+        key2 = parameters.generate_private_key()
+
+        shared_key_bytes = key2.exchange(key1.public_key())
+        symkey = int_from_bytes(shared_key_bytes, 'big')
+
+        symkey_manual = pow(key1.public_key().public_numbers().y,
+                            key2.private_numbers().x,
+                            parameters.parameter_numbers().p)
+
+        assert symkey == symkey_manual
+
+    def test_symmetric_key_padding(self, backend):
+        """
+        This test has specific parameters that produce a symmetric key
+        In length 63 bytes instead 64. We make sure here that we add
+        padding to the key.
+        """
+        p = int("11859949538425015739337467917303613431031019140213666"
+                "129025407300654026585086345323066284800963463204246390"
+                "256567934582260424238844463330887962689642467123")
+        g = 2
+        y = int("32155788395534640648739966373159697798396966919821525"
+                "72238852825117261342483718574508213761865276905503199"
+                "969908098203345481366464874759377454476688391248")
+        x = int("409364065449673443397833358558926598469347813468816037"
+                "268451847116982490733450463194921405069999008617231539"
+                "7147035896687401350877308899732826446337707128")
+        parameters = dh.DHParameterNumbers(p, g)
+        public = dh.DHPublicNumbers(y, parameters)
+        private = dh.DHPrivateNumbers(x, public)
+        key = private.private_key(backend)
+        symkey = key.exchange(public.public_key(backend))
+        assert len(symkey) == 512 // 8
+        assert symkey[:1] == b'\x00'
+
+    bad_tls_exchange_params = [(
```

Comment:
Could you move these values into a vector file (with a few small comments noting in what way each one is broken)?